test/src/test/java/jenkins/model/JenkinsTest.java

```diff
@@ -338,18 +341,22 @@ public void testDoEval() throws Exception {
             grant(Jenkins.ADMINISTER).everywhere().to("alice").
             grant(Jenkins.READ).everywhere().to("bob").
             grantWithoutImplication(Jenkins.ADMINISTER, Jenkins.READ).everywhere().to("charlie"));
+
         WebClient wc = j.createWebClient();
-        wc.login("alice");
+
+        wc.withBasicApiToken(User.getById("alice", true));
         wc.assertFails("eval", HttpURLConnection.HTTP_BAD_METHOD);
         assertEquals("3", eval(wc));
-        wc.login("bob");
+
+        wc.withBasicApiToken(User.getById("bob", true));
         try {
             eval(wc);
             fail("bob has only READ");
         } catch (FailingHttpStatusCodeException e) {
             assertEquals(HttpURLConnection.HTTP_FORBIDDEN, e.getStatusCode());
         }
-        wc.login("charlie");
+
+        wc.withBasicApiToken(User.getById("charlie", true));
```

Comment:
Maybe it would be better for `withBasicApiToken(String)` to just call `getById(name, true)`? If you really wanted to assert that the user record already existed (which for a typical test using `DummySecurityRealm` would be irrelevant), you could do so explicitly. It would simplify a bunch of code in this PR to make such a change.

Reply:
Initially my idea was that could lead to potential misunderstanding. We could change that behavior in a second PR in order to have this one merged.

Reply:
Agreed.